axelor-sale/src/main/java/com/axelor/apps/sale/web/SaleOrderLineController.java

```diff
@@ -456,4 +457,54 @@ public void translateProductDescriptionAndName(ActionRequest request, ActionResp
       TraceBackService.trace(response, e);
     }
   }
+
+  public void calculateRelatedSOLinesValues(ActionRequest request, ActionResponse response) {
+    try {
+
+      RelatedSaleOrderLineService relatedSaleOrderLineService =
+          Beans.get(RelatedSaleOrderLineService.class);
+      Context context = request.getContext();
+      SaleOrderLine saleOrderLine = context.asType(SaleOrderLine.class);
+      saleOrderLine = Beans.get(SaleOrderLineRepository.class).find(saleOrderLine.getId());
+      SaleOrder saleOrder = relatedSaleOrderLineService.getSaleOrderFromContext(context);
+      relatedSaleOrderLineService.updateRelatedSOLinesOnPriceChange(saleOrderLine, saleOrder);
+      relatedSaleOrderLineService.updateRelatedSOLinesOnQtyChange(saleOrderLine, saleOrder);
+      saleOrderLine.setIsProcessedLine(true);
+      response.setReload(true);
+
+    } catch (Exception e) {
+      TraceBackService.trace(response, e);
+    }
+  }
+
+  public void hideAti(ActionRequest request, ActionResponse response) {
+
+    Context context = request.getContext();
+    SaleOrder saleOrder = Beans.get(SaleOrderLineService.class).getSaleOrder(context);
+    response.setAttr("exTaxTotal", "hidden", saleOrder.getInAti());
+    response.setAttr("inTaxTotal", "hidden", !saleOrder.getInAti());
+    response.setAttr("price", "hidden", saleOrder.getInAti());
+    response.setAttr("inTaxPrice", "hidden", !saleOrder.getInAti());
+  }
+
+  public void setLineIndex(ActionRequest request, ActionResponse response) {
+
+    Context context = request.getContext();
+    SaleOrderLine saleOrderLine = context.asType(SaleOrderLine.class);
+    saleOrderLine =
+        Beans.get(RelatedSaleOrderLineService.class).setLineIndex(saleOrderLine, context);
+    response.setValues(saleOrderLine);
+  }
+
+  public void updateSaleOrderLineListSize(ActionRequest request, ActionResponse response) {
+    SaleOrderLine saleOrderLine = request.getContext().asType(SaleOrderLine.class);
+    saleOrderLine.setSaleOrderLineListSize(saleOrderLine.getSaleOrderLineList().size());
+    for (SaleOrderLine slo : saleOrderLine.getSaleOrderLineList()) {
+      if (slo.getIsProcessedLine() || slo.getIsDisabledFromCalculation()) {
+        saleOrderLine.setIsDisabledFromCalculation(true);
+        break;
+      }
+    }
+    response.setValues(saleOrderLine);
```

Comment:
Directly use setValue for 'saleOrderLineListSize' instead of using setValues to avoid issues.